conformance/lang/expressions/literals/int-literal.balt

```diff
@@ -0,0 +1,470 @@
+Labels: int
+
+public function main() {
+    int x = 101;
+    io:println(x); // @output 101
+
+    x = 0;
+    io:println(x); // @output 0
+
+    x = 3;
+    io:println(x); // @output 3
+
+    x = 9223372036854775807;
+    io:println(x); // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with function call.
+Labels: int
+
+public function main() {
+    printInt(101);
+
+    printInt(0);
+
+    printInt(3);
+
+    printInt(9223372036854775807);
+}
+
+function printInt(int x) {
+    io:println(x); // @output 101
+                   // @output 0
+                   // @output 3
+                   // @output 9223372036854775807
+}
+
+Test-Case: error
+Description: Test DecimalNumbers with leading zeros.
+Labels: int
+
+public function main() {
+    int x = 00; // @error invalid DecimalNumber with leading '0'
+
+    x = 012345; // @error invalid DecimalNumber with leading '0'
+}
+
+Test-Case: output
+Description: Test HexIntLiterals with int-type-descriptor and function call.
+Labels: int
+
+public function main() {
+    int x = 0x1;
+    printInt(x);
+
+    x = 0xa;
+    printInt(x);
+
+    x = 0xA;
+    printInt(x);
+
+    x = 0x000000a;
+    printInt(x);
+
+    x = 0x1fed12cba7;
+    printInt(x);
+
+    x = 0xABCDEF81;
+    printInt(x);
+
+    x = 0x192AAce;
+    printInt(x);
+
+    x = 0X1;
+    printInt(x);
+
+    x = 0Xa;
+    printInt(x);
+
+    x = 0XA;
+    printInt(x);
+
+    x = 0XABCDEF81;
+    printInt(x);
+
+    printInt(0X1fedcba7);
+
+    printInt(0X192AAc5);
+}
+
+function printInt(int x) {
+    io:println(x); // @output 1
+                   // @output 10
+                   // @output 10
+                   // @output 10
+                   // @output 137121418151
+                   // @output 2882400129
+                   // @output 26389198
+                   // @output 1
+                   // @output 10
+                   // @output 10
+                   // @output 2882400129
+                   // @output 535677863
+                   // @output 26389189
+}
+
+Test-Case: error
+Description: Test invalid HexInLiteral.
+Fail-Issue: ballerina-lang #32559
+Labels: int
+
+public function main() {
+    int x = 0x; // @error invalid HexIntLiteral due to absence of HexDigits
+
+    x = 0X; // @error invalid HexIntLiteral due to absence of HexDigits
+
+    x = 0x 12fa; // @error invalid HexIntLiteral
+
+    x = 0x123g; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 0X123G; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 123ab; // @error undefined symbol 'ab'
+}
+
+Test-Case: output
+Description: Test int-literals with toString().
+Labels: int
+
+public function main() {
+    io:println((101).toString()); // @output 101
+
+    io:println((0).toString()); // @output 0
+
+    io:println((9223372036854775807).toString()); // @output 9223372036854775807
+
+    io:println((0x12ab).toString()); // @output 4779
```

Comment:
These cases should work without parenthesis too. For some reason, `io:println(0x12ab.toString());` is not working. Others are working as expected. 

Shall we create an issue for the not working one?

I have missed it :( when adding tests with, https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-parser/src/test/resources/expressions/access-expr/method_call_expr_source_01.bal

Reply:
Okay. Will create an issue and update the PR.